### ISSUE TITLE
Storage Lectern with Mimic Scrolls doesn't work with locked, empty Functional Storage drawers

### ISSUE THREAD

**Issue by Ignoranus** [FTBTeam/FTB-Modpack-Issues#5195]
### Modpack

FTB Skies Expert

### Modpack version

1.7.0

### Describe the bug

Functional Storage drawers that are empty but locked to a specific item are not recognized by the Storage Lectern as the correct inventory for that item.

If there are still any items in the locked drawer, behavior is as expected: new items inserted into the Lectern get put into the drawer. However, when there are no items remaining in the drawer, the Storage Lectern treats the inventory as truly empty and any of the relevant item added to the storage network are sent to the more general (no scrolls attached) storages instead of to the drawer that's locked to that specific item.

### Steps to reproduce

1. Place an Ars Nouveau Storage Lectern and add at least one Bookwyrm charm
2. Place a storage drawer nearby
3. Attach an item frame to the drawer 
4. Insert an Ars Nouveau Mimic Scroll
5. Put any item in the storage drawer (e.g. oak logs)
6. Use the Configuration Tool to lock the storage drawer
7. Link the drawer to the Storage Lectern using the Dominion Wand (either directly or via a linked Storage Controller)
8. Place a chest and link it to the storage lectern (with no item frame/scroll attached)
9. Remove all items from the locked drawer (but leave it locked)
10. Put the items directly into the storage lectern (or into an adjacent unlinked 'input' chest)

The items will be added to the empty chest, rather than the locked 'mimic' storage drawer.

### Expected behaviour

Locked drawers with Mimic Scrolls should still get the relevant item from storage lecterns, even if empty

### Screenshots

![image](https://github.com/FTBTeam/FTB-Modpack-Issues/assets/108766037/13c9cefb-4a45-43f7-8efc-0d348a81b02f)

**Comment by botanySupremist** [comment]
I am unsure if this is a pack bug or an interaction between these mods, but this has been annoying me as well.

**Comment by Gaz492** [comment]
Is this still an issue for you in the latest version of the modpack?

**Comment by Ignoranus** [comment]
I did some more testing and the issue does still occur for me, but I have more specific details to provide:

I created a simple testing setup: several stacks of Stone, two storage drawers, a controller, a chest, and a Storage Lectern. The chest and one drawer are connected directly to the lectern, the remaining drawer is connected to the controller (which is connected to the lectern). I locked both drawers to "stone" using the configuration tool.

![image](https://github.com/FTBTeam/FTB-Modpack-Issues/assets/108766037/8896d1b5-6586-4815-9793-c77af3a19363)


It seems as if the Storage Lectern chooses where to put items based on the order that inventories were linked to the Lectern. Inventories linked first are tried first. In my testing setup, I could get the Stone to consistently go into the drawer as long as the following were met:


1. The drawer (including via storage controller) has no stone, but has room for stone, and no other linked inventories (regardless of linking order) have stone in them and room for more stone.

2. The drawer has stone and is earlier in the order than any other inventories that would be able to accept stone


Mimic scrolls don't seem to do anything when attached to storage drawers or storage controllers, so I'm not sure if I'm missing something completely there.

I would still expect that the Storage Lectern, particularly with a mimic scroll, should recognize that the storage drawer (or storage controller) is a suitable/preferred destination for the items in question, instead of treating it like a regular empty inventory slot, like the way AE2 seems to handle it.
